app/lib/lets-encrypt.server.ts

```diff
@@ -1,21 +1,79 @@
 import acme from 'acme-client';
-import type { Client as AcmeClient } from 'acme-client';
+import type {
+  Client as AcmeClient,
+  Order as AcmeOrder,
+  Authorization as AcmeAuthorization,
+} from 'acme-client';
+import type {
+  Account as AcmeAccount,
+  DnsChallenge as AcmeDnsChallenge,
+} from 'acme-client/types/rfc8555';
 import { secrets } from 'docker-secret';
 
+interface ChallengeBundle {
+  domain: string;
+  value: String;
+}
+
 const { LETS_ENCRYPT_ACCOUNT_PRIVATE_KEY_PEM } = secrets ?? {};
 
+/**
+ * Usage examples
+ *
+ * new LetsEncrypt()
+ *  .initialize()
+ *  .then((le) => le.createOrder("xyz.com"))
+ *  .then((le) => console.log(le.challengeBundles));
+ *
+ * new LetsEncrypt()
+ *  .initialize()
+ *  .then((le) => le.recallOrder(orderUrl))
+ *  .then((le) => console.log(le.challengeBundles));
+ */
+
 class LetsEncrypt {
   #client?: AcmeClient;
 
   #directoryUrl?: string;
 
   #accountKey?: string;
 
+  #account?: AcmeAccount;
+
+  #order?: AcmeOrder;
+
+  #authorizations?: AcmeAuthorization[];
+
+  #challengeBundles?: ChallengeBundle[];
+
+  /**
+   * This will reload the account based on the `accountKey`. If the key has no
+   * associated account, it will create a new one and return that
+   */
+  #registerReloadAccount = async () => {
+    // https://datatracker.ietf.org/doc/html/rfc8555#section-7.3.1
+
+    const account = await this.#client!.createAccount({
+      termsOfServiceAgreed: true,
+      contact: [`mailto:${process.env.LETS_ENCRYPT_ACCOUNT_EMAIL}`],
+    });
+
+    if (account.status !== 'valid') throw new Error(`Acme account is "${account.status}"`);
```

Comment:
I didn't know `prettier` allows `if` statement without curly brackets. To me, line break after `if` is more readable.

Reply:
I actually like to omit the code block for one liner statements that stop execution of the code (next, break, return, throw) I think it looks cleaner

Reply:
When I was working with Mozilla on Firefox and Gecko, we used to regularly see bugs happen due to code like this:

```js
// Original code
if(whatever)
  doSomething();
```

Someone would make a change, like so:

```js
// Modified code
if(whatever)
  doSomething();
  doSomethingElse();
```

This obviously breaks, but only some of the time!  Mozilla's engineers got to the point where they decided it wasn't worth the hassle of having people accidentally break code by forgetting to wrap multi-line blocks.

The rule became, "always brace your ifs" and I've done it ever since.  This pattern will never break:

```js
// Original code
if(whatever) {
  doSomething();
}

// Modified code
if(whatever) {
  doSomething();
  doSomethingElse();
}
```

You'd think, "no one will ever make this mistake!"  But I've seen some of the best devs in the world do it.  As you get older, you start to appreciate simple code practices more.

Reply:
I agree, and do exactly that for everything except for commands stopping the code execution (next, continue, break, return, throw).

If we have no strong convention, I'd keep using it as is, but if we decide that we want to prevent these kinds of errors, and braces are always mandatory, we should enforce it with an eslint rule:
https://eslint.org/docs/latest/rules/curly

As always, I'll yield to the convention if we decide to do this.

BTW, I'd also suggest adding this:
https://eslint.org/docs/latest/rules/no-console
In the very rare cases when we actually do want to use console, we can turn it off with `//eslint-disable-next-line no-console`

Reply:
https://github.com/Seneca-CDOT/starchart/issues/102